src/protocol/messageSet/decoder.js

```diff
@@ -64,7 +65,20 @@ const EntriesDecoder = (decoder, compressedMessage) => {
 }
 
 const EntryDecoder = decoder => {
+  if (!decoder.canReadInt64()) {
+    throw new KafkaJSPartialMessageError(
+      `Tried to decode a partial message: There isn't enough bytes to read the offset`
+    )
+  }
+
   const offset = decoder.readInt64().toString()
+
+  if (!decoder.canReadInt32()) {
```

Comment:
We should probably consider refactoring the decoder readInt* methods to do this check instead, so we don't keep running into this problem.

Reply:
Yeah, this solution isn't great but I'm still not seeing a better pattern here. If we move this to the decoder we lose the context since it's too deep in the implementation to understand that it's a partial message or something else. Maybe adding the boundary checks there and recover from this error everywhere? I'm not sure yet